File: SDKs/vmware.local/vrni/1.0.0/python/src/sdk/models/shared/basevirtualmachine.py
from dataclasses import dataclass, field
from typing import List,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class BaseVirtualMachine:
    default_gateway: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('default_gateway') }})
    destination_firewall_rules: Optional[List[RuleSet]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('destination_firewall_rules') }})
    entity_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('entity_id') }})
    entity_type: Optional[EntityTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('entity_type') }})
    ip_addresses: Optional[List[IPV4Address]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ip_addresses') }})
    ip_sets: Optional[List[Reference]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ip_sets') }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    security_groups: Optional[List[Reference]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('security_groups') }})
    source_firewall_rules: Optional[List[RuleSet]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('source_firewall_rules') }})
    vnics: Optional[List[Reference]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('vnics') }})
    

File: SDKs/vmware.local/vrni/1.0.0/typescript/src/sdk/models/shared/basevirtualmachine.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { RuleSet } from "./ruleset";
import { EntityTypeEnum } from "./entitytypeenum";
import { IpV4Address } from "./ipv4address";
import { Reference } from "./reference";



export class BaseVirtualMachine extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=default_gateway" })
  defaultGateway?: string;

  @SpeakeasyMetadata({ data: "json, name=destination_firewall_rules", elemType: RuleSet })
  destinationFirewallRules?: RuleSet[];

  @SpeakeasyMetadata({ data: "json, name=entity_id" })
  entityId?: string;

  @SpeakeasyMetadata({ data: "json, name=entity_type" })
  entityType?: EntityTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=ip_addresses", elemType: IpV4Address })
  ipAddresses?: IpV4Address[];

  @SpeakeasyMetadata({ data: "json, name=ip_sets", elemType: Reference })
  ipSets?: Reference[];

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=security_groups", elemType: Reference })
  securityGroups?: Reference[];

  @SpeakeasyMetadata({ data: "json, name=source_firewall_rules", elemType: RuleSet })
  sourceFirewallRules?: RuleSet[];

  @SpeakeasyMetadata({ data: "json, name=vnics", elemType: Reference })
  vnics?: Reference[];
}
